File: api-legacy/src/main/java/com/bitstachio/bazaarapi/user/service/UserService.java
package com.bitstachio.bazaarapi.user.service;

import com.bitstachio.bazaarapi.user.dto.UserCreateRequest;
import com.bitstachio.bazaarapi.user.dto.UserResponse;
import com.bitstachio.bazaarapi.user.dto.UserUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface UserService {

    UserResponse create(UserCreateRequest request);

    UserResponse getById(UUID id);

    List<UserResponse> getAll();

    UserResponse update(UUID id, UserUpdateRequest request);

    void delete(UUID id);
}

File: api/src/main/kotlin/dev/bitstachio/interdimensional_bazaar/user/service/UserService.kt
package dev.bitstachio.interdimensional_bazaar.user.service

import dev.bitstachio.interdimensional_bazaar.user.dto.UserCreateRequest
import dev.bitstachio.interdimensional_bazaar.user.dto.UserResponse
import dev.bitstachio.interdimensional_bazaar.user.dto.UserUpdateRequest
import java.util.UUID

interface UserService {
	fun create(request: UserCreateRequest): UserResponse

	fun getById(id: UUID): UserResponse

	fun getAll(): List<UserResponse>

	fun update(id: UUID, request: UserUpdateRequest): UserResponse

	fun delete(id: UUID)
}
